src/snes.jl

```diff
@@ -12,8 +12,11 @@ mutable struct SNES{T}
     update_jac!
     jac_A
     jac_P
+    julia_vec::Cint 
```

Comment:
Maybe use a boolean?
```suggestion
    use_julia_vec::Bool 
```